boomerang/src/lib.rs

```diff
@@ -9,17 +9,32 @@ pub mod icp_to_nicp;
 pub mod log;
 pub mod nicp_to_icp;
 
+#[cfg(test)]
+pub mod state_machine;
+
 // "ryjl3-tyaaa-aaaaa-aaaba-cai"
 pub const ICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 2, 1, 1]);
 
 // "buwm7-7yaaa-aaaar-qagva-cai"
+#[cfg(not(feature = "test-env"))]
 pub const NICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 48, 1, 170, 1, 1]);
+// "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+#[cfg(feature = "test-env")]
+pub const NICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 0, 1, 1]);
 
 // "jcmow-hyaaa-aaaaq-aadlq-cai"
+#[cfg(not(feature = "test-env"))]
 pub const WTN_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 0, 0, 215, 1, 1]);
+// "renrk-eyaaa-aaaaa-aaada-cai"
+#[cfg(feature = "test-env")]
+pub const WTN_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 6, 1, 1]);
 
 // "tsbvt-pyaaa-aaaar-qafva-cai"
+#[cfg(not(feature = "test-env"))]
 pub const WATER_NEURON_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 48, 1, 106, 1, 1]);
+// "r7inp-6aaaa-aaaaa-aaabq-cai"
+#[cfg(feature = "test-env")]
+pub const WATER_NEURON_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 3, 1, 1]);
 
```

Comment:
That's not the way to go if you want to go on this path create a state, but if we want to stay light you should create canisters at right ids in state machine, I think there is a method to do that but we have to check.

Reply:
"In information technology and computer science, a system is described as stateful if it is designed to remember preceding events or user interactions; the remembered information is called the state of the system. The set of states a system can occupy is known as its state space." wikipedia

Reply:
Our canister is not a state hence I came up with another solution to make it work as intended